clippy_lints/src/methods/mod.rs

```diff
@@ -3453,14 +3473,21 @@ fn lint_suspicious_map(cx: &LateContext<'_>, expr: &hir::Expr<'_>) {
     );
 }
 
+const OPTION_AS_REF_DEREF_MSRV: &str = "1.40.0";
+
```

Comment:
@flip1995 I couldn't have these constants as `semver::Version` objects as non-const method calls aren't allowed for initializing constants. Is there a better way to handle this?

Reply:
All of the fields of `Version` are public, so you can construct it as 
```rust
Version {
    major: 1,
    minor: 40,
    patch: 0,
    pre: Vec::new(),
    build: Vec::new(),
}
```
 while we're waiting on https://github.com/steveklabnik/semver/pull/216 to get merged and released.